src/services/types.ts

```diff
@@ -409,6 +410,16 @@ namespace ts {
         commands?: CodeActionCommand[];
     }
 
+    export interface CodeFix extends CodeAction {
+        /** If present, one may call 'getCombinedCodeFix' with this actionId. */
+        actionId?: {} | undefined;

```

Comment:
This is on a separate type for back compat reasons?  Making it optional isn't enough?

Reply:
A `CodeFix` includes both a `CodeAction` and an optional action id. A `CodeAction` will only have an `actionId` if returned from a code fix, not if from a completion.